sympy/matrices/expressions/matpow.py

```diff
@@ -138,3 +145,7 @@ def _eval_derivative_matrix_lines(self, x):
 
     def _eval_inverse(self):
         return MatPow(self.base, -self.exp)
+
+
+Pow._kind_dict[(NumberKind, MatrixKind)] = MatrixKind
+Pow._kind_dict[(MatrixKind, NumberKind)] = MatrixKind
```

Comment:
Would it make more sense to give MatPow a distinct `_kind_dict`?

This modifies the `_kind_dict` attribute of `Pow`...

Reply:
`MatPow` is not subclass of Pow.